### PR TITLE
Plugin Rename & Code refactor.

### PR DESCRIPTION
**Pull request by bahiirwa**
Hoping to close issues like #8, #5, #2, #4

- Rename the prefix
- Add new plugin text domain i18n
- Fix docs and Credits and Rename plugin
- Remove unused css and html
- Prefix all functions
- Add raw text as in issue #2 description.
- Properly enqueue the scripts #4 

Screenshot 2020-12-19 at 13 10 59

Screenshot 2020-12-19 at 13 11 11

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
I don't really know enough to assess this. Changes to readme look fine. No idea about the main php file.

**Comment by bahiirwa** [comment]
@simplycomputing Could you test this branch in a CP install to see if all works well? User experience should be the same as before. Nothing has fundamentally changed apart from having a txt file on server. 

Yah, you might have to make new snippets as the post type name has changed. Or you can migrate the old posts using a plugin like post type switcher by JJJ

**Comment by ghost** [comment]
The shortcode isn't working for me. Just shows the shortcode.

**Comment by ghost** [comment]
Ah, OK. You changed the shortcode text. Working now.

**Comment by ghost** [comment]
"View Raw" gives a 404. Also if I go to the code snippet page and try "view page" then it gives a 404 as well.

**Comment by bahiirwa** [comment]
Have you tried to save the settings as postname again on CP. post type things. 😄

**Review by timbocode — CHANGES_REQUESTED** [review]
This is good work!

I've not tested this exhaustively yet but a couple of quick comments:

1. It should be made obvious what the shortcode format is somewhere in admin, either in the code list page or on the code edit page. Or both.
2. I got a "page not found" when clicking on the "View Raw" button
3. When clicking on the "Copy to Clipboard" button, if the code contains special characters (such as `\n`), it does not paste correctly. For example:

**Original (copied) code:**
```
$notices = (array) preg_split( '~[\r\n]+~', trim( $matches[2] ) );
```

**Result when pasted:**
```
$notices = (array) preg_split( '~[
]+~', trim( $matches[2] ) );
```

and

**Original**

```
$upgrade_notice .= preg_replace( '~\[([^\]]*)\]\(([^\)]*)\)~', '<a href="${2}">${1}</a>', $line );
```

**Pasted**

```
$upgrade_notice .= preg_replace( '~[([^]]*)](([^)]*))~', '<a href="2">1</a>', $line );
```

The code seems to display fine. It's just when it's copied to the clipboard.

4. Can we stick with the original proposed name of "**Classic Code Snippets**" everywhere please ("Code Snippets" is fine for the admin menu). 

Other than that, it's looking really good. It will be a nice addition to the plugin ecosystem, so well done both.

**Comment by bahiirwa** [comment]
Thanks for the review. Will go test and extract the noted issues in depth.

For clarity
> It should be made obvious what the shortcode format is somewhere in admin, either in the code list page or on the code edit page. Or both.

Do you mean something like this in the snippet edit page? (I am borrowing from Gravity forms)



If yes, that would a new feature request to work on.

**Comment by timbocode** [comment]
Yes, something like the Gravity PDF doodah would be great. New feature request is fine but in the meantime, it would be good to add "example usage" to the plugin description, like so:

ccs-usage

**Comment by ghost** [comment]
Re-saving permalinks fixed the 404 issue (forgot about the need for that).

All looks good!

**Comment by timbocode** [comment]
Just to confirm, resaving permalinks sorted the 404 for me too. And the raw code displayed fine.

**Comment by bahiirwa** [comment]
I have looked into the other issues and resolved them.

> When clicking on the "Copy to Clipboard" button, if the code contains special characters (such as \n), it does not paste correctly.

Seem this issue is with the JS we are using. I will research further on this.

**Comment by ghost** [comment]
I am just testing this again. Is it true that the main file is still called classic-display-code.php?

Shouldn't this be changed to classic-code-snippets.php?

**Comment by timbocode** [comment]
There's some good work here @bahiirwa. 

Just to confirm @simplycomputing's comment about the main file. Yes, it should be called `classic-code-snippets.php` rather than `classic-display-code.php`.

Also need to look into the conflicting files. It might be happening because an earlier commit contained these two files (`form.php` and `sc-display-code.php`) but now neither of them exist in this latest commit. I'll try and look into it v soon.

**Comment by ghost** [comment]
Re formatting issues. I've just remembered that when I first set this up it did some weird things on the occasional snippet. I eventually worked out that it sort of auto-detects the type of code and formats accordingly. So mostly it correctly worked out it was php, but sometimes it would think it was something else. To fix this I set it to treat everything as php (because all of the CC snippets were php). You can see in the main file there is this:

```
return '
	<pre>
		<code class="language-php">' . $myVar  . '</code>
	</pre>
	<div>
```
That class may be causing issues and should be removed.

**Comment by bahiirwa** [comment]
My thoughts as well @simplycomputing. I will rebase and make a commit before review.

**Comment by ghost** [comment]
Trouble is, if I use it _without_ that class then the problem I was experiencing will return. I guess ideally it would have a selector so you could choose what language you want it to format to. 

That might be a possible enhancement for later.

**Comment by bahiirwa** [comment]
This can be reviewed as it. I have pulled out a number of issues I saw testing with the copy to clipboard implementation.

**Review by timbocode — CHANGES_REQUESTED** [review]
Looking good. Just a couple of additional observations:

1. The file `form.php` seems to be superfluous and should probably be removed.
2. When creating a new code snippet, I get the following PHP notice:

```
NOTICE: .../wp-content/plugins/classic-code-snippets/classic-code-snippets.php:128 - Undefined index: ccs_code_snippet
```

It's definitely getting there.

File: classic-code-snippets.php [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	<p class="ccs-meta-options">
+		<label for="ccs_code_snippet"><?php esc_attr_e( 'Insert snippet in post/page using', 'ccs-code-snippet'); ?></label>
+		<input id="ccs_code_snippet" type="text" name="ccs_code_snippet" value="<?php echo '[css_snippets id=' . get_the_ID() . ']'; ?>" />
+	</p>
+	<p class="ccs-meta-options">
+		<label for="ccs_code_snippet screen-reader-text"><?php echo esc_attr( 'Code Snippet', 'ccs-code-snippet'); ?></label><br>
+		<textarea id="ccs_code_snippet" rows="20" class="widefat" name="ccs_code_snippet"><?php echo esc_attr( $ccs_code_snippet ); ?></textarea>
+	</p>
+	<?php
+}
```
Comment (bahiirwa) [inline classic-code-snippets.php]: I agree @timbocode. This particular piece removes the `form.php` file.

I think it was done to separate the concerns of the technologies.

File: classic-code-snippets.php [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+function ccs_display_callback( $post ) {
+	$ccs_code_snippet = get_post_meta( get_the_ID(), 'ccs_code_snippet', true );
+	?>
+	<p class="ccs-meta-options">
+		<label for="ccs_code_snippet"><?php esc_attr_e( 'Insert snippet in post/page using', 'ccs-code-snippet'); ?></label>
+		<input id="ccs_code_snippet" type="text" name="ccs_code_snippet" value="<?php echo '[css_snippets id=' . get_the_ID() . ']'; ?>" />
+	</p>
+	<p class="ccs-meta-options">
+		<label for="ccs_code_snippet screen-reader-text"><?php echo esc_attr( 'Code Snippet', 'ccs-code-snippet'); ?></label><br>
+		<textarea id="ccs_code_snippet" rows="20" class="widefat" name="ccs_code_snippet"><?php echo esc_attr( $ccs_code_snippet ); ?></textarea>
```
Comment (bahiirwa) [inline classic-code-snippets.php]: My bad. I see that I left both input fields with the same name. That conflicts with the metadata as it is in a singular form. I will make the html ammends.

Comment (bahiirwa) [reply classic-code-snippets.php]: Fixed

**Comment by timbocode** [comment]
I seem to have broken it. My bad.

The shortcode no longer works for me. It just displays `[ccs_snippet id=xxx]` although I've also noticed that the shortcode name appears to have changed to `css_snippets` (with an 's' at the end) - which I've also tried.

I'm also still getting `PHP Notice:  Undefined index: ccs_code_snippet in .../classic-code-snippets/classic-code-snippets.php on line 128` when creating a new code post and also when creating a new, ordinary post.

I've tried on two separate sites (different themes etc.).

**Comment by bahiirwa** [comment]
> The shortcode no longer works for me. It just displays [ccs_snippet id=xxx] although I've also noticed that the shortcode name appears to have changed to css_snippets (with an 's' at the end) - which I've also tried.

I should be tried on the stake. Added a fix.

I have tried to follow these steps to recreate your issue.
1. Installed fresh CP and turned on `WP_DEBUG`in `wp-config.php` etc.
2. Pull plugin from repo.
3. `git co rename` branch
4. Activate plugin.
5. Create new snippet
6. Add snippet to page.
7. View page. I do not have issue raised.

**Comment by bahiirwa** [comment]
>I'm also still getting PHP Notice: Undefined index: ccs_code_snippet in .../classic-code-snippets/classic-code-snippets.php on line 128 when creating a new code post and also when creating a new, ordinary post.

However, with the last commit, I have rewritten the lines in 128 in the old way. Let me know if this is necessary and fixes the issue.

**Review by timbocode — APPROVED** [review]
All good now Laurence. Your fixes have worked.

The shortcode is working fine and I no longer get the PHP error.

So, all good to go now.

**Pull request merged**